src/Sentry/Internal/Http/CachingTransport.cs

```diff
@@ -310,13 +301,13 @@ await using (stream.ConfigureAwait(false))
         internal int GetCacheLength() => GetCacheFilePaths().Count();
 
         // This method asynchronously blocks until the envelope is written to cache, but not until it's sent
-        public async Task SendEnvelopeAsync(
+        public Task SendEnvelopeAsync(
             Envelope envelope,
             CancellationToken cancellationToken = default)
         {
             // Store the envelope in a file without actually sending it anywhere.
             // The envelope will get picked up by the background thread eventually.
-            await StoreToCacheAsync(envelope, cancellationToken).ConfigureAwait(false);
+            return StoreToCacheAsync(envelope, cancellationToken);
```

Comment:
It's best to await so we can debug this better by having this line in the stacktrace. Optimizing here doesn't pay off